File: Downloads/MemworCompany-master/MemworCompany-master/app/src/main/java/com/shurik/memwor_24/memwor/api/module_reddit/RedditResponse.java
package com.shurik.memwor_24.memwor.api.module_reddit;

import com.shurik.memwor_24.memwor.content.PostData;

import java.util.List;

public class RedditResponse {
    public RedditData data;

    public class RedditData {
        public List<RedditPost> children;

        public class RedditPost {
            public PostData data;
        }
    }
}

File: app/src/main/java/com/shurik/memwor_24/memwor/content/module_reddit/RedditResponse.kt
package com.shurik.memwor_24.memwor.content.module_reddit

data class RedditResponse(val data: Data)
data class Data(val children: List<Child>)
data class Child(val data: ChildData)
data class ChildData(
    val author: String,
    val subreddit: String,
    val selftext: String,
    val preview: Preview?
)

data class Preview(val images: List<Image>)
data class Image(val source: Source)
data class Source(val url: String)
